docs/development/decision-records/2024-08-28-Release-Workflow-Refactor/README.md

```diff
@@ -0,0 +1,83 @@
+# Refactor of the Release Workflow
+
+## Decision
+
+Refactor the existing workflows to accommodate the release of bugfix versions as described in [this](../2024-07-18-Improvements-versioning-scheme/README.md) decision record:
+- Remove the releases and release/X.Y.Z branches in favor of using just the main branch and tags.
+- Change the draft-new-release and publish-new-release to enable hotfix releases
+
+## Rationale
+
+As of this decision record creation date, the release of a new version is triggered with the execution of the `draft-new-release` workflow.
+The workflow branches from main and checks out into a `release/*` branch where a new commit to represent the releasable state is created. Afterward a PR is created to merge the release/X.Y.Z branch into `releases`.
+Once the PR is approved and merged, the `publish-new-release` workflow is automaticaly triggered to run the required steps in order to publish all artifacts
+in their desired repositories/platforms. With the successful publication of said artifacts, the head of the `releases` branch is merged back into `main` 
+to keep the integrity of the commit tree. Finally, the current project version is automatically bumped to the next minor version in a new commit to `main` 
+to kickstart the next development cycle.
+
+As they stand, the current release workflows don't allow the release of bugfix versions since:
+- the `draft-new-release` workflow doesn't create any bugfix branch;
+- even if the branch is manually created, merging it to `releases` via PR to trigger the `publish-new-release` workflow, is not possible;
+- The `publish-new-release` workflow can't be manually triggered;
+- Even if it could be manually triggered, there are steps that shouldn't apply for bugfix releases such as bumping the project version or merging to main;
+
+
+## Approach
+
+The draft-new-release workflow should include two extra inputs:
+- A `choice` input to allow the distinction between a release or bugfix.
```

Comment:
Though about this as well. Having to set the input would make the process more transparent for anyone using it and that is the only reason I added it like this. I'm fine with both ways.

Reply:
I think having both could be more prone-error, also because afaik the workflow dispatch input cannot be a "select" between two choice but just an open string. 
A sort of "documentation" can be added to the description of the `version` input to warn who's releasing that if the branch is main this will be a proper release, bugfix otherwise.

Reply:
Just fyi, you can actually have a choice input between two specific options:
![image](https://github.com/user-attachments/assets/fb04d749-9a6e-4f4a-a34e-6483457aa4a3)

However thinking more about it having the choice doesn't add any more benefits than the one i mentioned above, which a better description on the manual input window would suffice. I'll remove the choice thing.

Reply:
cool! I wasn't aware, thanks for the hint